packages/agent/src/agent/http/http.test.ts

```diff
@@ -253,3 +262,95 @@ describe('getDefaultFetch', () => {
     // TODO: Figure out how to test the self and default case errors
   });
 });
+
+describe('invalidate identity', () => {
+  const mockFetch: jest.Mock = jest.fn();
+  it('should allow its identity to be invalidated', () => {
+    const identity = new AnonymousIdentity();
+    const agent = new HttpAgent({ identity, fetch: mockFetch, host: 'http://localhost' });
+    const invalidate = () => agent.invalidateIdentity();
+    expect(invalidate).not.toThrowError();
+  });
+  it('should throw an error instead of making a call if its identity is invalidated', async () => {
+    const canisterId: Principal = Principal.fromText('2chl6-4hpzw-vqaaa-aaaaa-c');
+    const identity = new AnonymousIdentity();
+    const agent = new HttpAgent({ identity, fetch: mockFetch, host: 'http://localhost' });
+    agent.invalidateIdentity();
+
+    const expectedError =
+      "This identity has expired due this application's security policy. Please refresh your authentication.";
+
+    // Test Agent.call
+    await agent
+      .call(canisterId, {
+        methodName: 'test',
+        arg: new ArrayBuffer(16),
+      })
+      .catch((reason: AgentError) => {
+        expect(reason.message).toBe(expectedError);
+      });
+    // Test Agent.query
+    await agent
+      .query(canisterId, {
+        methodName: 'test',
+        arg: new ArrayBuffer(16),
+      })
+      .catch((reason: AgentError) => {
+        expect(reason.message).toBe(expectedError);
+      });
+    // Test readState
+    await agent
+      .readState(canisterId, {
+        paths: [[new ArrayBuffer(16)]],
+      })
+      .catch((reason: AgentError) => {
+        expect(reason.message).toBe(expectedError);
+      });
+  });
+});
+describe('replace identity', () => {
+  const mockFetch: jest.Mock = jest.fn();
+  it('should allow an actor to replace its identity', () => {
+    const identity = new AnonymousIdentity();
+    const agent = new HttpAgent({ identity, fetch: mockFetch, host: 'http://localhost' });
+
+    const identity2 = new AnonymousIdentity();
+    const replace = () => agent.replaceIdentity(identity2);
+    expect(replace).not.toThrowError();
```

Comment:
could you check that the new identity is the new one?

```
expect(agent._identity).toEqual(identity2);
```

Reply:
No, the identity is private out of security concerns. It is also cumbersome to try to inspect it out of the mockFetch payload

Reply:
ok!